File: app/src/main/java/com/github/ihsg/demo/util/SharedPreferencesUtil.java
package com.github.ihsg.demo.util;

import android.content.SharedPreferences;
import android.preference.PreferenceManager;

import com.github.ihsg.demo.TestApplication;

/**
 * Created by hsg on 14/10/2017.
 */

public class SharedPreferencesUtil {
    private static SharedPreferencesUtil instance;

    private SharedPreferences.Editor editor;
    private SharedPreferences prefer;

    public SharedPreferencesUtil() {
        this.prefer = PreferenceManager.getDefaultSharedPreferences(TestApplication.getContext());
        this.editor = this.prefer.edit();
    }

    public static SharedPreferencesUtil getInstance() {
        if (instance == null) {
            synchronized (SharedPreferencesUtil.class) {
                if (instance == null) {
                    instance = new SharedPreferencesUtil();
                }
            }
        }

        return instance;
    }

    public void saveString(String name, String data) {
        this.editor.putString(name, data);
        this.editor.commit();
    }

    public String getString(String name) {
        return this.prefer.getString(name, null);
    }
}

File: app/src/main/java/com/github/ihsg/demo/util/SharedPreferencesUtil.kt
package com.github.ihsg.demo.util

import android.content.SharedPreferences
import com.github.ihsg.demo.TestApplication

/**
 * Created by hsg on 14/10/2017.
 */
internal class SharedPreferencesUtil private constructor() {
    private val prefer: SharedPreferences = TestApplication.getContext()
        .getSharedPreferences("test", android.content.Context.MODE_PRIVATE)
    private val editor: SharedPreferences.Editor = prefer.edit()

    fun saveString(name: String?, data: String?) {
        editor.putString(name, data)
        editor.commit()
    }

    fun getString(name: String?): String? {
        return prefer.getString(name, null)
    }

    companion object {
        @JvmStatic
        var instance: SharedPreferencesUtil? = null
            get() {
                if (field == null) {
                    synchronized(SharedPreferencesUtil::class.java) {
                        if (field == null) {
                            field = SharedPreferencesUtil()
                        }
                    }
                }
                return field
            }
            private set
    }
}